test/contract_tests.js

```diff
@@ -0,0 +1,543 @@
+        })
+
+        it("updateReport(): Updating the non-attested report should revert", async function () {
+            await expect(instance.connect(Triager).updateReport(F_SUBMIT(report1), newStatusBitField)).to.be.revertedWith("Bug Report Notary: Report is unattested");
+        })
+
+    })
+
+    describe("===> disclose()",function(){
+        let getReportRoot,
+            merkleProofval,
+            commit,
+            rr,
+            kk,
+            key,
+            salt,
+            value;
+
+        this.beforeEach(async () => {
+            getReportRoot = F_SUBMIT(report1)
+            await instance.connect(Triager).submit(getReportRoot);
+            
+            [rr, kk, commit] = F_attest(report1, Triager.address)
+            
+            key = ReportKeys.report;
+            [getReportRoot, salt, value, merkleProofval] = F_disclose(report1,key);
+        });
+
+        it("disclose(): Anyone can disclose the report",async function(){
+            await instance.connect(Triager).attest(rr, kk, commit)
+
+            await expect(instance.connect(Triager).disclose(getReportRoot, key, salt, value, merkleProofval))
+            .to.emit(instance,'ReportDisclosure')
+            .withArgs(getReportRoot,key,value);
+        })
+
+        // [BUG] should fail : https://github.com/immunefi-team/notary/issues/16
+        it("disclose(): Doing Attest on Disclosed Report should revert",async function(){
+            await instance.connect(Triager).disclose(getReportRoot, key, salt, value, merkleProofval)
+            await expect(instance.connect(Triager).attest(getReportRoot, key, commit)).to.be.reverted;
+        })
+
+        // [BUG] should fail : https://github.com/immunefi-team/notary/issues/16
+        it("disclose(): Doing Update on Disclosed Report should revert", async function () {
+            await instance.connect(Triager).attest(getReportRoot, key, commit)
+            
+            await instance.connect(Triager).disclose(getReportRoot, key, salt, value, merkleProofval)
+            // false
+            //console.log("STATUS : ", await instance.connect(Triager).reportHasStatus(getReportRoot, Triager.address, 0))
+            await expect(instance.connect(Triager).updateReport(getReportRoot, 00000001)).to.be.reverted;;
+            // true
+            //console.log("STATUS : ", await instance.connect(Triager).reportHasStatus(getReportRoot, Triager.address, 0))
+        })
+
+        //[BUG] should fail : ASK DUNCAN
+        it("disclose(): Disclosing the already disclosed report should revert", async function () {
+            await instance.connect(Triager).attest(rr, kk, commit)
+
+            await expect(instance.connect(Triager).disclose(getReportRoot, key, salt, value, merkleProofval))
+            await expect(instance.connect(Triager).disclose(getReportRoot, key, salt, value, merkleProofval)).to.be.reverted;
+                //.to.be.revertedWith("Bug Report Notary: Key already disclosed for report");
```

Comment:
We changed the semantics to `disclose`. It doesn't revert if you call it twice. (But it doesn't log or update the timestamp on the 2nd call.)